deeplab/src/utils.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import * as tf from '@tensorflow/tfjs';
+import { config } from './config';
+import { Color, DeepLabInput, SemanticSegmentationBaseModel } from './types';
+
+const pascalColormapMaxEntriesNum = config['DATASET_MAX_ENTRIES']['PASCAL'];
+const createPascalColormap = (): Color[] => {
+  /**
+   * Generates the colormap matching the Pascal VOC dev guidelines.
+   * The original implementation in Python: https://git.io/fjgw5
+   */
+
+  const colormap = new Array(pascalColormapMaxEntriesNum);
+  for (let idx = 0; idx < pascalColormapMaxEntriesNum; ++idx) {
+    colormap[idx] = new Array(3);
+  }
+  for (let shift = 7; shift > 4; --shift) {
+    const indexShift = 3 * (7 - shift);
+    for (let channel = 0; channel < 3; ++channel) {
+      for (let idx = 0; idx < pascalColormapMaxEntriesNum; ++idx) {
+        colormap[idx][channel] |=
+          ((idx >> (channel + indexShift)) & 1) << shift;
+      }
+    }
+  }
+  return colormap;
+};
+
+const createADE20KColormap = (): Color[] => {
+  return config['COLORMAPS']['ADE20K'] as Color[];
+};
+
+const createCityscapesColormap = (): Color[] => {
+  return config['COLORMAPS']['CITYSCAPES'] as Color[];
+};
+
+export const getColormap = (base: SemanticSegmentationBaseModel) => {
+  if (base === 'pascal') {
```

Comment:
This will break the modularity of the code and make it less maintainable, since   the pascal colormap is recreated, not read from the config. I prefer keeping this as a utility function.